### ISSUE TITLE
Terminal rending fails under fractional scaling

### ISSUE THREAD

**Issue by andyczerwonka** [alacritty/alacritty#7598]
Only top left renders. Once I resize, it's fine.

![image](https://github.com/alacritty/alacritty/assets/138898/b62c6a53-78ba-42c5-bc8f-ff129974a411)

This only happens with fractional scaling.

![image](https://github.com/alacritty/alacritty/assets/138898/9a5b12ed-2c05-42e8-9b08-cabe523768e6)

### System

OS: Linux
Version: `alacritty 0.14.0-dev (5685ce8b)
Linux: Wayland, Mutter

### Logs

```
[0.000008475s] [INFO ] [alacritty] Welcome to Alacritty
[0.000149599s] [INFO ] [alacritty] Version 0.14.0-dev (5685ce8b)
[0.000167937s] [INFO ] [alacritty] Running on Wayland
[0.000549233s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/aczerwon/.config/alacritty/alacritty.toml"
[0.286894086s] [INFO ] [alacritty] Using EGL 1.5
[0.287001009s] [DEBUG] [alacritty] Picked GL Config:
                                     buffer_type: Some(Rgb { r_size: 8, g_size: 8, b_size: 8 })
                                     alpha_size: 8
                                     num_samples: 0
                                     hardware_accelerated: true
                                     supports_transparency: Some(true)
                                     config_api: Api(OPENGL | GLES1 | GLES2 | GLES3)
                                     srgb_capable: true
[0.325729198s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Medium, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.331563633s] [INFO ] [alacritty] Window scale factor: 1
[0.333103673s] [DEBUG] [alacritty] Loading "Fira Code" font
[0.335962037s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.341135283s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.345191371s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.347980294s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.351444590s] [INFO ] [alacritty] Running on Mesa Intel(R) Graphics (RPL-S)
[0.351474982s] [INFO ] [alacritty] OpenGL version 4.6 (Core Profile) Mesa 23.3.0-1pop0~1702935939~22.04~67e417a~dev, shader_version 4.60
[0.351482826s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.357627949s] [DEBUG] [alacritty] Enabled debug logging for OpenGL
[0.357726343s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.367843009s] [INFO ] [alacritty] Cell size: 10 x 21
[0.367873221s] [INFO ] [alacritty] Padding: 5 x 5
[0.367880560s] [INFO ] [alacritty] Width: 1210, Height: 640
[0.367918237s] [INFO ] [alacritty] PTY dimensions: 30 x 120
[0.370819221s] [INFO ] [alacritty] Initialisation complete
[0.390188551s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.392568106s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.394876566s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.397213397s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.398155691s] [INFO ] [alacritty] Font size changed to 34.666668 px
[0.398165903s] [INFO ] [alacritty] Cell size: 20 x 42
[0.398175782s] [DEBUG] [alacritty_terminal] New num_cols is 59 and num_lines is 14
[0.406538499s] [INFO ] [alacritty] Padding: 10 x 10
[0.406569802s] [INFO ] [alacritty] Width: 1210, Height: 640
```

**Comment by kchibisov** [comment]
Fixed on master and 0.13.1

**Comment by andyczerwonka** [comment]
Thanks @kchibisov Confirmed that this is fixed on master `0.14.0-dev (1b9131c7)`
